mindsdb/integrations/handlers/openai_handler/openai_handler.py

```diff
@@ -237,3 +243,127 @@ def describe(self, attribute: Optional[str] = None) -> pd.DataFrame:
 
         return pd.DataFrame([[meta['id'], meta['object'], meta['owned_by'], meta['permission'], args]],
                             columns=['id', 'object', 'owned_by', 'permission', 'model_args'])
+
+    def update(self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        """
+        Fine-tune OpenAI GPT models. Steps are roughly:
+          - Analyze input data and modify it according to suggestions made by the OpenAI utility tool
+          - Get a training and validation file
+          - Determine base model to use
+          - Submit a fine-tuning job via the OpenAI API
+          - Monitor progress with exponential backoff (which has been modified for greater control given a time budget in hours), 
+          - Gather stats once fine-tuning finishes
+          - Modify model metadata so that the new version triggers the fine-tuned version of the model (stored in the user's OpenAI account)
+
+        Caveats: 
+          - As base fine-tuning models, OpenAI only supports the original GPT ones: `ada`, `babbage`, `curie`, `davinci`. This means if you adjust successively more than once, any fine-tuning other than the most recent one is lost.
+        """  # noqa
+
+        args = args if args else {}
+        using_args = args.pop('using') if 'using' in args else {}
+        prompt_col = using_args.get('prompt_column', 'prompt')
+        completion_col = using_args.get('completion_column', 'completion')
+
+        for col in [prompt_col, completion_col]:
+            if col not in set(df.columns):
+                raise Exception(f"To fine-tune this OpenAI model, please format your select data query to have a `{prompt_col}` column and a `{completion_col}` column first.")  # noqa
+
+        args = {**using_args, **args}
+        prev_model_name = self.base_model_storage.json_get('args').get('model_name', '')
+
+        openai.api_key = self._get_api_key(args)
+        folder_name = 'finetune_artifacts'
+        adjust_time = datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+
+        temp_storage_path = self.engine_storage.folder_get(folder_name)
+        temp_file_name = f"ft_{adjust_time}"
+        temp_model_storage_path = f"{temp_storage_path}/{temp_file_name}.jsonl"
```

Comment:
Are iles in temp_storage_path temporal or have to be stored in engine storage?

Reply:
As discussed with @ea-rus, this should be a temp folder instead.